main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2873,6 +2874,14 @@ object Classpaths {
         }
     }
   }
+
+  def shellPromptFromState: State => String = { s: State =>
+    val extracted = Project.extract(s)
+    (name in extracted.currentRef).get(extracted.structure.data) match {
```

Comment:
Interesting, never noticed the get on `DefinableSetting`. This is equivalent to `extracted getOpt (name in extracted.currentRef)`.